ffmpeg/extras.c

```diff
@@ -163,6 +188,13 @@ int lpms_get_codec_info(char *fname, pcodec_info out)
   if(!audio_present && !video_present) {
     // instead of returning -1
     ret = GET_CODEC_STREAMS_MISSING;
+  } else if (video_present) {
+    out->dur = ic->duration / AV_TIME_BASE;
+  } else if (audio_present && !video_present) {
+    out->dur = calculate_duration(ic, astream);
```

Comment:
Why do we calculate duration only if video is not present?

Reply:
During development I found that `out->dur = ic->duration / AV_TIME_BASE;` was returning a very large negative number that wasn't useful. The `calculate_duration` function is a more general, but less efficient means of calculating the duration. It seemed logical to keep this only to audio-only. The function name is a bit ambiguous, should I rename it?

Reply:
Yeah, better to rename it. Plus, maybe add this comment into the code.

Reply:
Renamed `calculate_duration` to `calculate_audio_dur` and added the comment